algorithms/linfa-lars/src/lib.rs

```diff
@@ -0,0 +1,95 @@
+use linfa::{Float, traits::PredictInplace};
+use ndarray::{Array1, Array2, ArrayBase, Data, Ix2};
+
+pub use error::LarsError;
+pub use hyperparams::{LarsParams, LarsValidParams};
+
+mod algorithm;
+mod error;
+mod hyperparams;
+
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+/// Least angle regression a.k.a. LAR
```

Comment:
You can move the general documentation at the top of `linfa-lars` lib with the references. 
You can check what is done for [`linfa-pls`](https://github.com/rust-ml/linfa/blob/db63e70102f67c6a4313ea9f1cba8527841cb966/algorithms/linfa-pls/src/lib.rs#L1)